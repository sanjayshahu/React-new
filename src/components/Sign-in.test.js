import { shallow,mount,render } from "enzyme";
import { idText } from "typescript";
import {React}from "react";
import SignIn from "./Sign-in";
import toJson from 'enzyme-to-json';
it('take snapshot',()=>{
    expect( toJson(shallow(<SignIn/>))).toMatchSnapshot();//use this command for taking snapshot
}) 

it('check increment',()=>{
    const wrapper=shallow(<SignIn/>);//use this command for taking snapshot
    wrapper.find('[id="counter"]').simulate('click');
    expect(wrapper.state()).toEqual({count:1,email:"",password:''});
    wrapper.find('[id="counter"]').simulate('click')
    expect(wrapper.state()).toEqual({count:2,email:"",password:''});
    wrapper.find('[id="counter"]').simulate('keypress')//does nothing
    expect(wrapper.state()).toEqual({count:2,email:"",password:''});
    // expect(wrapper.props()).toEqual({count:2,email:"",password:''});
}) 
