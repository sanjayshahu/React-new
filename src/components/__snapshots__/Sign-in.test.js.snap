// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`take snapshot 1`] = `
<div
  className="sign-in"
>
  <h2>
    I already have an account
  </h2>
  <span>
    Sign in with your email and password
  </span>
  <form
    onSubmit={[Function]}
  >
    <FormInput
      handleChange={[Function]}
      label="email"
      name="email"
      required={true}
      type="email"
      value=""
    />
    <FormInput
      handleChange={[Function]}
      label="password"
      name="password"
      required={true}
      type="password"
      value=""
    />
    <div
      className="buttons"
    >
      <CustomButton
        id="counter"
        onClick={[Function]}
      >
         Increment 
        0
         
      </CustomButton>
      <CustomButton
        type="submit"
      >
         Sign in 
      </CustomButton>
      <CustomButton
        isGoogleSignIn={true}
        onClick={[Function]}
      >
        Sign in with Google
      </CustomButton>
    </div>
  </form>
</div>
`;
